### ISSUE TITLE
Do not commit .DS_Store

### ISSUE THREAD

**Issue by sholladay** [kjchoi10/noirdoor-react#2]
`.DS_Store` is a [junk file](https://github.com/sindresorhus/junk/blob/master/index.js) that doesn't add any value to the project, but does add bloat and causes messy git diffs.

To fix this:

1. `find . -name .DS_Store -print0 | xargs -0 git rm --ignore-unmatch`
2. `echo .DS_Store >> ~/.gitignore`
3. `git config --global core.excludesFile ~/.gitignore`

The above commands ignore it globally rather than locally because `.DS_Store` is not generated by anything related to a specific repository.

**Comment by kjchoi10** [comment]
Question:
I don't understand why we didn't add in the issue #1.

 `git config --global core.excludesFile ~/.gitignore`

It seems like this git config command makes git ignore anything in the gitignore file for any future git pushes. Does that mean I should always include this command anytime I make a new repo and want to push code into it from my mac computer?

**Comment by sholladay** [comment]
The difference in the commands stems from _what kind of files_ we are talking about.

In #1, we are talking about `node_modules`, something specific only to this being a Node.js project. You might very well also be a Python developer and you don't want to pollute your Python projects with references to `node_modules` in their `.gitignore` files. The `.gitignore` file within a repo should be a minimal representation of what kind of files might be generated during development of that individual project _and_ that are specific to each developer (e.g. test results).

In #2, here we are talking about something _global_ to you being on macOS. It is specific to your machine and specifically **not** related to the project itself. You want this to apply to all repos you'll ever have on that machine. Also, you don't want to put a reference to `.DS_Store` in the repo's `.gitignore` because all of your other collaborators might be using Windows and it is just bloat and irrelevant for them.

Basically, we are categorizing these types of files / directories and putting them in an appropriate place depending on whose responsibility it is.

There's actually 3 different kinds of ignore files. Here they are in order of precedence from highest to lowest.

| Type | Path | Commit & Push | Responsibility |
| ----- | ---- | ---- | ---- |
| Local public | `<repo>/.gitignore` | Yes | Project-generated files. |
| Local private | `<repo>/.git/info/exclude` | N/A | Files specific to you **and** the project. |
| Global | `$HOME/.gitignore` | N/A | OS junk files, IDE files, etc. |
